mirrormaker_replication_flow.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2018 Aiven, Helsinki, Finland. https://aiven.io/
+
+package aiven
+
+type (
+	// MirrorMakerReplicationFlowHandler is the client which interacts with the
+	// Kafka MirrorMaker 2 ReplicationFlows endpoints on Aiven.
+	MirrorMakerReplicationFlowHandler struct {
+		client *Client
+	}
+
+	// MirrorMakerReplicationFlowRequest request used to create a Kafka MirrorMaker 2
+	// ReplicationFlows entry.
+	MirrorMakerReplicationFlowRequest struct {
+		Enabled         bool     `json:"enabled"`
+		SourceCluster   string   `json:"source_cluster"`
+		TargetCluster   string   `json:"target_cluster"`
+		Topics          []string `json:"topics"`
+		TopicsBlacklist []string `json:"topics.blacklist"`
+	}
+
+	// MirrorMakerReplicationFlowResponse represents the response from Aiven after
+	// interacting with the Kafka MirrorMaker 2 API.
+	MirrorMakerReplicationFlowResponse struct {
+		APIResponse
+		ReplicationFlows MirrorMakerReplicationFlowRequest `json:"replication_flows"`
+	}
+)
+
+// Create creates new Kafka MirrorMaker 2 Replication Flows entry.
+func (h *MirrorMakerReplicationFlowHandler) Create(project, service string, req MirrorMakerReplicationFlowRequest) (*MirrorMakerReplicationFlowResponse, error) {
+	path := buildPath("project", project, "service", service, "mirrormaker", "replication-flows")
+	bts, err := h.client.doPostRequest(path, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var rsp MirrorMakerReplicationFlowResponse
+	if errR := checkAPIResponse(bts, &rsp); errR != nil {
+		return nil, errR
+	}
+
+	return &rsp, nil
+}
+
+// Update updates new Kafka MirrorMaker 2 Replication Flows entry.
+func (h *MirrorMakerReplicationFlowHandler) Update(project, service string, req MirrorMakerReplicationFlowRequest) (*MirrorMakerReplicationFlowResponse, error) {
+	path := buildPath("project", project, "service", service, "mirrormaker", "replication-flows")
+	bts, err := h.client.doPutRequest(path, req)
+	if err != nil {
+		return nil, err
+	}
+
+	var rsp MirrorMakerReplicationFlowResponse
+	if errR := checkAPIResponse(bts, &rsp); errR != nil {
+		return nil, errR
+	}
+
+	return &rsp, nil
+}
+
+// Get gets a Kafka MirrorMaker 2 Replication Flows.
+func (h *MirrorMakerReplicationFlowHandler) Get(project, service string) (*MirrorMakerReplicationFlowResponse, error) {
+	path := buildPath("project", project, "service", service, "mirrormaker", "replication-flows")
+	bts, err := h.client.doGetRequest(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var rsp MirrorMakerReplicationFlowResponse
+	if errR := checkAPIResponse(bts, &rsp); errR != nil {
+		return nil, errR
+	}
+
+	return &rsp, nil
+}
+
+// Delete deletes a Kafka MirrorMaker 2 Replication Flows entry.
+func (h *MirrorMakerReplicationFlowHandler) Delete(project, service string) error {
+	path := buildPath("project", project, "service", service, "mirrormaker", "replication-flows")
```

Comment:
Server doesn't implement delete for this path at all.

Reply:
then it looks like we have an outdated documentation https://api.aiven.io/doc/#api-Service_-_Kafka_MirrorMaker-ServiceKafkaMirrorMakerDeleteReplicationFlow

Reply:
The documentation looks valid. There is `<source_cluster>/<target_cluster>` at the end of the path, which is missing from here.

Reply:
Ah indeed will update the code